smack-extensions/src/test/java/org/jivesoftware/smackx/nick/NickManagerTest.java

```diff
@@ -0,0 +1,57 @@
+/**
+ *
+ * Copyright Miguel Hincapie
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.nick;
+
+import org.jivesoftware.smack.DummyConnection;
+import org.jivesoftware.smackx.InitExtensions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotNull;
+
+
+public class NickManagerTest extends InitExtensions {
```

Comment:
@Flowdalic I was searching in current tests and I couldn't find anything as a guide or to follow, the most close to it was DeliveryReceiptTest, so please be patient with this part :).

Can I use Mockito?
i.e. for getInstanceFor I wish to:
- test if `INSTANCES` is really working.
- test if addSyncStanzaListener is being called inside constructor.

Reply:
I'm feeling lonely with this u_u'